### PR TITLE
Template email renderer

### PR DESCRIPTION
**Pull request by PoulpiFr**

### CODE REVIEW

**Comment by PoulpiFr** [comment]
Hehe, that's the 2 helpers that I forgot when I wrote the plugin
implementation. Hence I didn't properly test them.

I'll fix both issues.

Le mar. 19 juin 2018 à 22:25, David Duponchel <notifications@github.com> a
écrit :

File: src/mediarithmics/api/plugin/emailtemplaterenderer/EmailInterface.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 export interface PluginEmailMeta {
-  from_email: string;
-  from_name: string;
-  to_email: string;
-  to_name: string;
-  reply_to: string;
-  subject_line: string;
+  from_email?: string;
```
Comment (dduponchel) [inline src/mediarithmics/api/plugin/emailtemplaterenderer/EmailInterface.ts]: Is it a backward compatible change ?

File: src/extra/handlebars/HandlebarsEngine.ts [hunk 2]
```diff
@@ -150,11 +167,60 @@ export class HandlebarsEngine
...
-  compile(template: string) {
+  enableProfileDataLayer(): void {
+
+    this.engine.registerHelper("profileData", function (opts: ProfileDataHelperOptions) {
+
+      // See https://blog.osteele.com/2007/12/cheap-monads/
+      const $N = {};
+      // Check if we have the value in the DataLayer
+      if ((((((opts || $N).data || $N).root || $N).private || $N).profileData || $N)[opts.hash.fieldName]) {
+        return opts.data.root.private.profileData[opts.hash.fieldName];
```
Comment (dduponchel) [inline src/extra/handlebars/HandlebarsEngine.ts]: That looks awfully close to a vendor specific implementation.

File: src/mediarithmics/api/plugin/emailtemplaterenderer/EmailInterface.ts (hunk 1 above)
Comment (PoulpiFr) [reply src/mediarithmics/api/plugin/emailtemplaterenderer/EmailInterface.ts]: Turning something mandatory into something optional is always backwards compatible, isn't it?

File: src/extra/handlebars/HandlebarsEngine.ts (hunk 2 above)
Comment (PoulpiFr) [reply src/extra/handlebars/HandlebarsEngine.ts]: Do you want me to remove it?

At first, it look terrible, but when writing it/reading it after the facts, it's not such a nightmare.

**Pull request merged**